readme.txt
=== Responsi WP Login Page ===
Contributors: a3rev, a3rev Software
Tags: responsi, responsi framework, wordpress login page
Requires at least: 4.4
Tested up to: 4.9.4
Stable tag: 1.4.1
License: GPLv3
License URI: http://www.gnu.org/licenses/gpl-3.0.html

The Responsi WP Login Page allows admin to edit the login page via the customizer menu 

== Description ==
Install and activate the Responsi WP Login Page and it adds WP Login Page menu to the Pages menu on the customizer.

== Features ==

* Create a custom design for your login page via the WordPress customizer
* WP Login page shows in the customizer when you access the WP Login Page Menu
* Customize the login page while you are logged in as admin


== Installation ==
Minimum Requirements
* WordPress 4.4
* PHP version 5.5 or greater
* MySQL version 5.5 or greater


== Usage ==
1. Install Responsi WP Login Page plugin from the Appearance > Add-on Plugins menu and activate it
2. Go to Pages > WP Login Page menu and start customizing the login page that shows in the preview


== Changelog ==

= 1.0 - 2016/01/22 =
* Initial Release - Commit Plugins source code to a3rev Github repository.
* Initial Release - Configure to work with Responsi Add-on Plugins Free plugin menu
* Initial Release - Commit Plugin to Amazon Cloud storage for download.





changelog.txt
*** Responsi WP Login Page ***

= 1.4.1 - 2023/01/13 =
* The maintenance release adds style for the Wordfence 2FA code box.
* Tweak - Apply CSS for the Wordfence 2FA form code entry.

= 1.4.0 - 2022/03/29 =
* This security hardening release follows a full PHPCS code audit with code refactoring and security hardening including additional escaping and sanitizing.
* Security - Escape all $-variable
* Security - Sanitize all $_REQUEST, $_GET, $_POST
* Security - Apply wp_unslash before sanitizing

= 1.3.12 - 2022/01/25 =
* This is a maintenance release for compatibility with WordPress major version 5.9 and Responsi 8.3.1 
* Tweak - Test for compatibility with WordPress 5.9
* Tweak - Test for compatibility with Responsi 8.3.1

= 1.3.11 - 2021/07/19 =
* This maintenance release is for compatibility with WordPress major version 5.8
* Tweak - Test for compatibility with WordPress 5.8
* Tweak - Add Update URI to plugin header with unique name
* Tweak - Skip version 1.3.10 to prevent PHP version misread

= 1.3.9 - 2021/02/02 =
* This maintenance release fixes a jQuery bug preventing the site logo showing on the login page
* Fix - jQuery not working with WordPress 5.6

= 1.3.8 - 2020/04/02 =
* This maintenance release is for compatibility with WordPress 5.4 plus 2 PHP fixes
* Tweak - Tested for compatibility with WordPress major version 5.4
* Tweak - Configure and run Travis CI unit test builds on the frameworks github repo.
* Fix - Update global ${$this- to $GLOBALS[$this to resolve 7.0+ PHP warnings
* Fix - Update global ${$option to $GLOBALS[$option to resolve 7.0+ PHP warnings

= 1.3.7 - 2019/12/16 =
* This feature upgrade is a major refactor of the PHP code to Composer Dependency Manager, a security review and hardening, compatibility with Responsi 7.7.0 and WordPress 5.3.1.
* Feature - Plugin fully refactored to Composer for cleaner and faster PHP code
* Tweak - Tested for compatibility with Responsi version 7.7.0
* Tweak - Tested for compatibility with WordPress version 5.3.1
* Dev - Ensure that all inputs are sanitized
* Dev - Ensure that all outputs are escaped.

= 1.3.6 - 2019/09/03 =
* This upgrade is for compatibility with Responsi Framework 7.6.0
* Refactor – Rewrite frontend HTML, CSS, JS and Variables Class and ID names to the new framework 7.6.0 namespaces 
* Refactor – Rewrite customizer HTML, CSS, JS and Variables Class and ID names to the new framework 7.6.0 namespaces
* Refactor - HTML, CSS, JS and Variables Class and ID names called from the framework making plugin file sizes smaller and faster to load

= 1.3.5 - 2019/07/25 =
* This upgrade is for compatibility with Responsi Framework 7.5.0
* Refactor - All plugin CSS and JS from Class ID to Class name for new Responsi Class names.

= 1.3.4 - 2019/04/02 =
* This maintenance update is for compatibility with WordPress Major version 5.1.1 plus a customizer menu bug fix when using blank child theme.
* Tweak - Tested for compatibility with WordPress 5.1.1
* Fix - Not saving customizer menu options when using the Blank Child Theme

= 1.3.3 - 2018/08/06 =
* This maintenance update is for compatibility with WordPress 4.9.8, Gutenberg 3.4.0, PHP 7.1.0, PHP 7.2.0, Responsi Framework 7.0.2
* Tweak - Tested for compatibility with Framework v 7.0.2
* Tweak - Tested for compatibility with WordPress 4.9.8
* Tweak - Tested for compatibility with Gutenberg 3.4.0 
* Tweak - Replace create_function deprecated in PHP 7.2.0

= 1.3.2 - 2018/06/27 =
* This Maintenance update is a refactor of customizer code for compatibility with Responsi framework version 6.9.6 
* Refactor - Conversion of single border and border radius controls to a group control for improved customizer menu UI and UX
* Refactor - Customizer for new Partial save function from Responsi framework. Only save what has been changed not save all. 
* Tweak - Compatibility with Responsi Framework version 6.9.6
* Tweak - Test for compatibility with WordPress 4.9.6
* Fix - Default options save in customizer.

= 1.3.1 - 2018/03/13 =
* Maintenance Update. This update follows a full security review and extensive security hardening  
* Tweak - Sanitized all database inputs and escaped functions 

= 1.3.0 - 2018/02/13 =
* Feature Upgrade - This upgrade features full integration with newly launched a3rev Dashboard plus compatibility with WordPress 4.9.4
* Framework - Update Check Now feature for compatibility with a3rev Dashboard 2.3.0
* Tweak - Test for compatibility with WordPress 4.9.4

= 1.2.4 - 2017/12/15 =
* Major maintenance update. Completely rewrite of the customizer controls in Customizer JS API for Compatibility with Responsi v 6.8.2. Don’t upgrade this plugin before you upgrade the framework.
* Tweak - Refactor of customize controls from using PHP API to use Customizer JS API WP 4.9
* Tweak - Elimination of Repeated ID in controls
* Tweak - Major cleanup of framework HTML on customizer controls to speed up customizer load time
* Tweak - Tested for compatibility with WordPress 4.9.1
* Tweak - Tested for compatibility with Responsi Framework 6.8.2

= 1.2.3 - 2017/11/16 =
* Maintenance Upgrade. Please upgrade to WordPress 4.9.0 – then Responsi Framework to 6.8.0 and then this plugin
* Tweak - Tested for compatibility with WordPress Major version 4.9
* Tweak - Updated for full compatibility with Responsi Framework new feature version 6.8.0

= 1.2.2 - 2017/04/22 =
* Tweak - Tested for full compatibility with WordPress 4.7.4
* Tweak - Tested for full compatibility with WooCommerce 3.0.4
* Fix - Remove get_users on filter-responsi-template-developer to remove duplicate queries when developer plugin admin menu is opened in browser

= 1.2.1 - 2016/11/22 =
* Tweak - Compile less mincss to inline style compatibility with Responsi Framework version 6.5.9.4 
* Tweak - Tested for full compatibility with Responsi Framework major version 6.5.9.4
* Tweak - Tested for full compatibility with WordPress major version 4.7.0

= 1.2.0 - 2016/10/26 =

responsi-wp-login-page.php
<?php
/*
Plugin Name: Responsi WP Login Page
Description: Don't want your client / customers seeing the WordPress logo and links when they login to your site? Responsi WP Login Page allows you to customize the login page to match your sites design and remove all reference to WordPress.
Version: 1.4.1
Author: a3rev Software
Author URI: https://a3rev.com/
Update URI: a3-responsi-wp-login-page
Requires at least: 4.4
Tested up to: 5.9
Text Domain: responsi-wp-login-page
Domain Path: /languages
License: This software is distributed under the terms of GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007

	Responsi WP Login Page. Plugin for the Responsi Framework.
	Copyright © 2011 a3THEMES

	a3THEMES
	admin@a3rev.com
	PO Box 1170
	Gympie 4570
	QLD Australia
*/

// File Security Check
if ( ! defined( 'ABSPATH' ) ) exit;
define( 'RESPONSI_WPLOGIN_PATH', dirname(__FILE__));
define( 'RESPONSI_WPLOGIN_FOLDER', dirname(plugin_basename(__FILE__)) );
define( 'RESPONSI_WPLOGIN_NAME', plugin_basename(__FILE__) );
define( 'RESPONSI_WPLOGIN_URL', str_replace( array( 'http:','https:' ), '', untrailingslashit( plugins_url( '/', __FILE__ ) ) ) );
define( 'RESPONSI_WPLOGIN_IMAGES_URL', RESPONSI_WPLOGIN_URL . '/assets/images' );
define( 'RESPONSI_WPLOGIN_JS_URL', RESPONSI_WPLOGIN_URL . '/assets/js' );
define( 'RESPONSI_WPLOGIN_CSS_URL', RESPONSI_WPLOGIN_URL . '/assets/css' );
define( 'RESPONSI_WPLOGIN_KEY', 'responsi_wp_login_page' );
define( 'RESPONSI_WPLOGIN_VERSION', '1.4.1' );

function responsi_wp_login_page_activate_validate() {
    if ( 'responsi' !== get_template() ) {
        echo sprintf( wp_kses_post( 'This is a plugin for Responsi Framework, you need to install <a href="%s" target="_blank" rel="noopener">Responsi Framework</a> theme from WordPress first before can activate this.', 'responsi-wp-login-page' ), 'https://wordpress.org/themes/responsi-framework/' );
        die();
    }
    update_option('a3rev_responsi_wp_login_page_version', RESPONSI_WPLOGIN_VERSION );
    update_option('responsi_wp_login_page_installed', true);
}

register_activation_hook(__FILE__,'responsi_wp_login_page_activate_validate');

if( !defined( 'RESPONSI_WPLOGIN_TRAVIS' ) ){
	if ( !file_exists( get_theme_root().'/responsi/functions.php' ) ) return;
	if ( !isset( $_POST['wp_customize'] ) && get_option('template') != 'responsi' ) return;
	if ( isset( $_POST['wp_customize'] ) && wp_unslash($_POST['wp_customize']) == 'on' && isset( $_POST['theme'] ) && stristr( wp_unslash($_POST['theme']), 'responsi' ) === FALSE ) return;
	if ( version_compare(get_option('responsi_framework_version'), '6.9.5', '<') ) return;
}

if ( version_compare( PHP_VERSION, '5.6.0', '>=' ) ) {
    require __DIR__ . '/vendor/autoload.php';

    global $responsi_wp_login_page_admin,$responsi_wp_login_page;
    $responsi_wp_login_page_admin                   = new \A3Rev\RWPLogin\Admin();
    $responsi_wp_login_page                         = new \A3Rev\RWPLogin\Main();
                                                      new \A3Rev\RWPLogin\Customizer();

} else {
    return;
}

add_action( 'after_setup_theme', 'responsi_addon_wp_login_page_upgrade_version' );
function responsi_addon_wp_login_page_upgrade_version() {

	if ( version_compare(get_option('a3rev_responsi_wp_login_page_version'), '1.4.1') === -1 ) {
		global $responsi_wp_login_page;
        $responsi_wp_login_page->build_css_after_addon_updated();
	}
	
	update_option('a3rev_responsi_wp_login_page_version', RESPONSI_WPLOGIN_VERSION );
}

include ( 'upgrade/plugin_upgrade.php' );
include ( 'admin/responsi-wp-login-page-init.php' );
include ( 'classes/responsi-wp-login-page-function.php' );

?>